platforms/software/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishIssuesIntegTest.groovy

```diff
@@ -450,4 +450,46 @@ subprojects {
         then:
         succeeds "help"
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/27569")
+    void "maven-publish is compatible with project isolation"() {
```

Comment:
This test states it verifies maven-publish is compatible with project isolation, but it does not actually turn on project isolation. 

IOW, this test still passes even without the changes to `DefaultMavenPublication`.

Is there a way to write this test to ensure it triggers a project isolation failure when executed without your changes to the production code?

Reply:
I run the tests using `:maven:isolatedProjectsIntegTest`, which failed before and passed after the change, and I also created a local Gradle installation with my real project, finally passing (also including signing).

I absolut get your point about the test name though, but I don't know your internal test setup to trigger isolated projects, the only workaround I know is opt-in PI using gradle.properties. Or do you prefer a better solution?

Reply:
Found a better solution using `@Requires(value = IntegTestPreconditions.IsolatedProjects)`. WDYT?

Reply:
Looking at this further, I don't really think its necessary to add a new test here at all. All of the tests in `MavenPublishMultiProjectIntegTest` already fail with this same error, as they also create publications from non-root projects. 

`MavenPublishMultiProjectIntegTest` already gives us the coverage we need to test this. I would just revert except for the actual production code changes in `DefaultMavenPublication` and then this LGTM

Reply:
Sure, will do it. But curios question, if the MavenPublishMultiProjectIntegTest fails, how does it pass the build?

Reply:
Isolated Projects is still in development. We do not run IP tests on CI. There are plenty of other tests in `:maven` that fail when IP is enabled.

Reply:
Currently, we run all tests as embedded, forking, and CC-on. Eventually as IP becomes more mature we will likely add an IP execution as well but for now so many things would fail that we are not ready yet

Reply:
For example it seems like using any kind of project dependency fails with IP enabled. See the tests in `MavenPublishJavaIntegTest` which cause failures with IP enabled.

Reply:
Oh, interesting... My very simple test project only contains external dependencies so I didn't notice it.